src/modules/user/index.ts

```diff
@@ -260,7 +260,7 @@ async function getUserPreference(token: any, baseURL: string, userPrefTypeId: st
     if (hasError(resp)) {
       throw resp.data
     }
-    return Promise.resolve(resp.data.userPrefValue ? JSON.parse(resp.data.userPrefValue) : {});
```

Comment:
Removing the JSON parsing will break the flow where we are using this method for fetching values.
Instead of this we should create a common method that checks whether the returned value can be parsed using JSON or not, if not return the value directly otherwise return the parsed JSON.